src/test/java/com/harragan/battleshipsboot/service/PlayerServiceTest.java

```diff
@@ -0,0 +1,58 @@
+package com.harragan.battleshipsboot.service;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import static org.mockito.Mockito.*;
+
+import com.harragan.battleshipsboot.model.game.GameArena;
+import com.harragan.battleshipsboot.model.game.Player;
+import com.harragan.battleshipsboot.repositorys.PlayerRepository;
+
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class PlayerServiceTest {
+    private PlayerService playerService;
+
+    @Before
+    public void initTest() {
+        playerService = new PlayerService(playerRepository);
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Mock
+    private PlayerRepository playerRepository;
+
+
+    @Test
+    public void playersCanBeAssignedAGameArenaWithAnId() {
+        GameArena gameArena = new GameArena();
+        Player player = new Player();
+        player.setId(1);
+        playerService.setArenaToPlayer(gameArena, player);
+        assertThat(player.getGameArena()).isEqualTo(gameArena);
+    }
+
+    @Test
+    public void givenWhenAPlayerIsInstantiatedAPlayerIdIsReturned() {
```

Comment:
The naming of this test could be adjusted to describe the behaviour better  eg. whenAPlayerIsInstantiatedThenAPlayerIdIsReturned